ocs_ci/deployment/flexy.py

```diff
@@ -0,0 +1,274 @@
+        # blocked till container terminates
+        with ThreadPoolExecutor(max_workers=1) as tp:
+            future = tp.submit(self.get_log)
+            cp = subprocess.run(
+                cmd_string, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+            )
+            if cp.returncode:
+                if cp.stderr:
+                    logger.error(cp.stderr)
+                raise exceptions.CommandFailed("Flexy run failed")
+            logger.info("Flexy run finished successfully")
+            self.exitThread = True
+        for res in TimeoutSampler(60, 1, future.done):
+            if not res:
+                logger.info("Waiting for logger thread to join")
+
+    def build_install_cmd(self):
+        """
+        Build flexy command line for 'deploy' operation
+
+        """
+        cmd = "sudo podman run --rm=true"
+        flexy_container_args = self.build_container_args()
+        cmd_string = " ".join([cmd, " ".join(flexy_container_args)])
+        return cmd_string
+
+    def build_destroy_cmd(self):
+        """
+        Build flexy command line for 'destroy' operation
+
+        """
+        cmd = "sudo podman run --rm=true"
+        flexy_container_args = self.build_container_args()
+        cmd_string = " ".join([cmd, " ".join(flexy_container_args), 'destroy'])
+        return cmd_string
+
+    def get_log(self):
+        """
+        This function runs in a seperate thread and continuously reads
+        logs from flexy podman
+
+        """
+        for res in TimeoutSampler(20, 1, os.path.exists, self.flexy_log_file):
+            if res:
+                break
+        log_fd = open(self.flexy_log_file, "r")
+
+        while True:
+            line = log_fd.readline()
+            if line:
+                logger.info(line.strip())
+            if self.exitThread:
+                break
+        logger.info("Logger thread done printing flexy logs")
+        # exit only this thread
+        sys.exit()
+
+    def build_container_args(self):
+        """
+        Builds most commonly used arguments for flexy container
+
+        """
+        args = list()
+        args.append(f"--log-opt={self.flexy_log_file}")
+        args.append(f"--env-file={self.flex_env_file}")
+        args.append(f"-w={self.flexy_mnt_container_dir}")
+        args.append(
+            f"--mount=type=bind,source={self.flexy_mnt_host_dir},"
+            f"destination=f{self.flexy_mnt_container_dir},relabel=shared"
+        )
+        args.append(
+            f"--mount=type=bind,source={self.flexy_host_private_conf_dir_path},"
+            f"destination={self.flexy_mnt_private_conf_dir},relabel=shared"
+        )
+        args.append(f"{self.flexy_img_url}")
+        return args
+
+    def clone_and_unlock_ocs_private_conf(self):
+        """
+        Clone ocs_private_conf (flexy env and config) repo into
+        flexy_mnt_host_dir
+
+        """
+        clone_repo(self.flexy_private_conf_url, self.flexy_host_private_conf_dir_path)
```

Comment:
We were discussing with @dahorak that we would like to:
1) Have possibility to change those info by config which you can already define URL. But we are missing also the custom branch. So please have variable also for branch name.

2) The repository does not have to be cloned and you can use just env file from local source as Daniel pointed out in different comment maybe. So this way we can have variable for env file passed also via config, and if defined it will have higher priority and then we don't need to clone the private repository via OCS-CI at all and just reuse specific file?

3) We thought that maybe will be better to use multiple env files for different approaches like PSI / Disconnected on AWS and so on, and change just specific variables we need instead of using only one env file.  @dahorak can you maybe elaborate more on this maybe on proper place and drop a comment there. Thanks

Reply:
My points are, that different templates might require different variables to be set and it might be quite messy to configure them all through the normal *ocs-ci* configuration files. And also it might be good to have good overview with what configuration the particular run was executed, but on the other hand some of the information in the env file might be considered as private, so it is probably not good to print it to the log. If it will be directly the env file from private git repository with only few minor changes (e.g. cluster name, OCP image version), than anybody with the rights to access the git repo can review the configuration but it will not be "leaked" widely.

Reply:
done, @petr-balogh one thing I couldn't address is private-conf branch. Flexy conf file doesn't have option to get particular branch but only URL for the repo can be specified in env file.

Reply:
@shylesh I think that you can by using this variable:  GIT_PRIVATE_BRANCH .  Flexy is processing variable names and it should work similar like with repo: `GIT_PRIVATE_OPENSHIFT_MISC_URI` where branch can be defined by just have _BRANCH at the end of variable instead of _URI.  Like:  `GIT_PRIVATE_OPENSHIFT_MISC_BRANCH`

Reply:
@petr-balogh Ok I will try this manually, Daniel also told the same thing about branch . I will update my findings.